### PR TITLE
Add a C-API "rb_hash_clear"

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: spec/ruby/optional/capi/hash_spec.rb [hunk 1]
```diff
@@ -91,6 +91,14 @@
     end
   end
 
+  describe "rb_hash_clear" do
+    it "returns self that cleared keys and values" do
+      hsh = { key: 'value' }
```
Comment (YorickPeterse) [inline spec/ruby/optional/capi/hash_spec.rb]: Could you use the hash rocket syntax here? We'll probably want to backport this to our 1.8 branch which IIRC doesn't support the new symbol syntax.

Comment (kachick) [inline spec/ruby/optional/capi/hash_spec.rb]: @YorickPeterse Sure. I'll update with hash rocket :)

Comment (YorickPeterse) [inline spec/ruby/optional/capi/hash_spec.rb]: Thanks!

**Comment by YorickPeterse** [comment]
Thanks!

**Comment by kachick** [comment]
Thanks to merge!

**Pull request merged**